test/test_respondent_controller.py

```diff
@@ -2032,3 +2035,126 @@ def _enroll_respondent(self, session):
+        respondent_2.first_name = "Two"
+
+        respondent_3 = Respondent()
+        respondent_3.email_address = "test3@example.com"
+        respondent_3.first_name = "Three"
+
+        respondents_to_delete = [respondent_1, respondent_2, respondent_3]
+
+        mock_query = MagicMock()
+        mock_query.filter(Respondent.mark_for_deletion).return_value = respondents_to_delete
+        # To iterate over "for respondent in respondents" the __iter__ method needs defining for the Mock
+        mock_query.filter.return_value.__iter__.return_value = iter(respondents_to_delete)
+        mock_session.query.return_value = mock_query
+
+        mock_logger = MagicMock()
+        mock_bind.return_value = mock_logger
+
+        # Execute
+        with self.app.app_context():
+            delete_respondents_marked_for_deletion.__wrapped__(mock_session)
+
+        # Verify
+        expected_confirmation_email_calls = [
+            call("test1@example.com", "One"),
+            call("test2@example.com", "Two"),
+            call("test3@example.com", "Three"),
+        ]
+        mock_send_account_deletion_confirmation_email.assert_has_calls(expected_confirmation_email_calls)
+        self.assertEqual(mock_session.commit.call_count, 3)
+        mock_logger.info.assert_called_with("Respondent record deletions complete", failed_deletion_count=0)
+        mock_session.rollback.assert_not_called()
+
+    @patch(
+        "ras_party.controllers.respondent_controller.send_account_deletion_confirmation_email", new_callable=MagicMock
+    )
+    @patch("ras_party.controllers.respondent_controller._delete_respondent_records")
+    @patch("ras_party.controllers.respondent_controller.session", new_callable=MagicMock)
+    @patch("ras_party.controllers.respondent_controller.logger.bind", new_callable=MagicMock)
+    def test_delete_respondents_continues_when_exception(
+        self, mock_bind, mock_session, mock_delete_respondent_records, mock_send_account_deletion_confirmation_email
+    ):
+        # Setup
+        respondent_1 = Respondent()
+        respondent_1.email_address = "test1@example.com"
+        respondent_1.first_name = "One"
+        respondent_1.id = 1
+        respondent_1.party_uuid = "5bbd2117-e457-4ff7-9248-ff800bbd16c8"
+
+        respondent_2 = Respondent()
+        respondent_2.email_address = "test2@example.com"
+        respondent_2.first_name = "Two"
+        respondent_2.id = 2
+        respondent_2.party_uuid = "3be3b957-eb6d-4dec-ab9c-4b618a9e95d6"
+
+        respondent_3 = Respondent()
+        respondent_3.email_address = "test3@example.com"
+        respondent_3.first_name = "Three"
+        respondent_3.id = 3
+        respondent_3.party_uuid = "a2a03e6f-89b3-49c2-af82-4bd411ef3307"
+
+        respondents_to_delete = [respondent_1, respondent_2, respondent_3]
+
+        mock_query = MagicMock()
+        mock_query.filter(Respondent.mark_for_deletion).return_value = respondents_to_delete
```

Comment:
This looks wrong to me and can be removed

Reply:
Removed, was left over from a previous approach.